### PR TITLE
Improve `FunctionTestCase` error message

### PR DESCRIPTION
**Pull request by niboshi**
#### Previous
```
E   chainer.testing.function.FunctionTestError: Parameterized test failed.
E   
E   Base test method: TestReLU_use_chainerx_false__chainerx_device_None__use_cuda_false__cuda_device_None__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never.test_forward
E   Test parameters:
E     contiguous: None
E     dtype: <class 'numpy.float16'>
E     shape: (3, 2)
E   
E   
E   (caused by)
E   FunctionTestError: Outputs of forward() do not match the expected values.
E   Indices of outputs that do not match: 0
E   Expected shapes and dtypes: (3, 2):float16
E   Actual shapes and dtypes:   (3, 2):float16
```

#### This PR
```
E   chainer.testing.function.FunctionTestError: Parameterized test failed.
E   
E   Base test method: TestReLU_use_chainerx_false__chainerx_device_None__use_cuda_false__cuda_device_None__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never.test_forward
E   Test parameters:
E     contiguous: None
E     dtype: <class 'numpy.float16'>
E     shape: (3, 2)
E   
E   
E   (caused by)
E   FunctionTestError: Outputs of forward() do not match the expected values.
E   Indices of outputs that do not match: 0
E   Expected shapes and dtypes: (3, 2):float16
E   Actual shapes and dtypes:   (3, 2):float16
E   
E   
E   Error details of output [0]:
E   
E   Not equal to tolerance rtol=0.0001, atol=1e-05
E   
E   Mismatch: 100%
E   Max absolute difference: 1.
E   Max relative difference: 1.
E    x: array([[0.0868, 0.    ],
E          [0.    , 0.6895],
E          [0.    , 0.    ]], dtype=float16)
E    y: array([[1.087  , 0.5566 ],
E          [0.849  , 1.689  ],
E          [0.00928, 0.2432 ]], dtype=float16)
E   
E   assert_allclose failed: 
E     shape: (3, 2) (3, 2)
E     dtype: float16 float16
E     i: (0, 0)
E     x[i]: 0.0867919921875
E     y[i]: 1.0869140625
E     relative error[i]: 0.919921875
E     absolute error[i]: 1.0
E   x: [[0.0868 0.    ]
E       [0.     0.6895]
E       [0.     0.    ]]
E   y: [[1.087   0.5566 ]
E       [0.849   1.689  ]
E       [0.00928 0.2432 ]]
```

### CODE REVIEW

File: tests/chainer_tests/functions_tests/activation_tests/test_relu.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def generate_inputs(self):
 
     def forward(self, inputs, device):
         x, = inputs
-        return functions.relu(x),
+        return functions.relu(x + 1),
```
Comment (toslunar) [inline tests/chainer_tests/functions_tests/activation_tests/test_relu.py]: (TODO: revert the change to the line)

Comment (niboshi) [reply tests/chainer_tests/functions_tests/activation_tests/test_relu.py]: Oops...

Comment (niboshi) [reply tests/chainer_tests/functions_tests/activation_tests/test_relu.py]: Fixed.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
The Jenkins failures are unrelated to the PR.
